frontend/src/pages/modelServing/screens/projects/NIMServiceModal/DeployNIMServiceModal.tsx

```diff
@@ -156,11 +159,18 @@ const DeployNIMServiceModal: React.FC<DeployNIMServiceModalProps> = ({
   React.useEffect(() => {
     const fetchNIMServingRuntimeTemplate = async () => {
       const nimTemplate = await getNIMServingRuntimeTemplate(dashboardNamespace);
-      setServingRuntimeSelected(getServingRuntimeFromTemplate(nimTemplate));
+      if (nimTemplate) {
+        const servingRuntime = getServingRuntimeFromTemplate(nimTemplate);
+        setServingRuntimeSelected(
+          servingRuntime ? updatePVCName(servingRuntime, nimPVCName) : undefined,
+        );
+      } else {
+        setServingRuntimeSelected(undefined);
+      }
     };
```

Comment:
Hmmm -- I think it's an interesting thought to modify it directly as you get it -- you can keep this if you want. I'd have thought to do the action on submit.

If you keep this action, generate the PVC name part of the effort -- using `useState` without the setter function being used is always a bit odd to see; I imagine there are good use-cases for it -- but I'm not sure it matters unless you submit.

Reply:
Updated